src/useIsShake.ts

```diff
@@ -51,14 +54,41 @@ export function useIsShake() {
     setSubscription(null);
   };
 
+  const checkAccelerometerAvail = async () => {
+    const availability = await isAvailableAsync();
+    setIsAccelerometerAvail(() => {
+      setIsAccelerometerAvailPending(false);
+      return availability;
+    });
+  };
```

Comment:
First check to see if Accelerometer is available to app.

Reply:
If it is only ever going to check if it is available on initialization, add "initialized" or "init" in it somewhere in the function name. Perhaps `checkAccelerometerAvailOnInit`